src/main/java/io/logz/log4j2/LogzioAppender.java

```diff
@@ -263,37 +295,80 @@ public void start() {
-            bufferDirPath = bufferDir;
-            File bufferFile = new File(bufferDirPath);
-            if (bufferFile.exists()) {
-                if (!bufferFile.canWrite()) {
-                    statusLogger.error("We cant write to your bufferDir location: "+bufferFile.getAbsolutePath());
+
+        if (!inMemoryBuffer) {
+            if (!(fileSystemFullPercentThreshold >= 1 && fileSystemFullPercentThreshold <= 100)) {
+                if (fileSystemFullPercentThreshold != -1) {
+                    statusLogger.error("fileSystemFullPercentThreshold should be a number between 1 and 100, or -1");
                     return;
                 }
-            } else {
-                if (!bufferFile.mkdirs()) {
-                    statusLogger.error("We cant create your bufferDir location: "+bufferFile.getAbsolutePath());
+            }
+
+            String bufferDirPath;
+            if (bufferDir != null) {
+                bufferDirPath = bufferDir;
+                File bufferFile = new File(bufferDirPath);
+                if (bufferFile.exists()) {
+                    if (!bufferFile.canWrite()) {
+                        statusLogger.error("We cant write to your bufferDir location: " + bufferFile.getAbsolutePath());
+                        return;
+                    }
+                } else {
+                    if (!bufferFile.mkdirs()) {
+                        statusLogger.error("We cant create your bufferDir location: " + bufferFile.getAbsolutePath());
+                        return;
+                    }
+                }
+            }
+            else {
+                bufferDirPath = System.getProperty("java.io.tmpdir") + File.separator+"logzio-log4j2-buffer";
+            }
+            File bufferDirFile = new File(bufferDirPath, logzioType);
+            tasksExecutor = Executors.newScheduledThreadPool(3, Log4jThreadFactory.createDaemonThreadFactory(this.getClass().getSimpleName()));
+            try {
+                logzioSender = LogzioSender
+                        .builder()
+                        .setDebug(debug)
+                        .setTasksExecutor(tasksExecutor)
+                        .setDrainTimeout(drainTimeoutSec)
+                        .setReporter(new StatusReporter())
+                        .setHttpsRequestConfiguration(conf)
+                        .WithDiskMemoryQueue()
+                          .setBufferDir(bufferDirFile)
+                          .setFsPercentThreshold(fileSystemFullPercentThreshold)
+                          .setGcPersistedQueueFilesIntervalSeconds(gcPersistedQueueFilesIntervalSeconds)
+                        .EndDiskQueue()
+                        .build();
+            } catch (LogzioParameterErrorException e) {
+                statusLogger.error("Some of the configuration parameters of logz.io is wrong: "+e.getMessage(),e);
+                return;
+            }
+        }else {
+            if (bufferSizeThreshold <= 0) {
```

Comment:
why not:
if(bufferSizeThreshold <= 0 && bufferSizeThreshold != -1){
                    statusLogger.error("bufferSizeThreshold should be a non zero integer, or -1");
                    return;
}